WinNSTabView.m

```diff
@@ -0,0 +1,228 @@
+            }
+        }
+    }
+  
+  return size.height;
+}
+
+- (void) drawTabViewRect: (NSRect)rect
+                  inView: (NSView *)view
+               withItems: (NSArray *)items
+            selectedItem: (NSTabViewItem *)selected
+{
+  if (!IsThemeActive())
+    {
+      [super drawTabViewRect: rect
+                      inView: view
+                   withItems: items
+                selectedItem: selected];
+    }
+  else
+    {
+      // Ensure the requested drawing rectangle includes the tab area...
+      const NSTabViewType type      = [(NSTabView *)view tabViewType];
+      const CGFloat       tabHeight = [self tabHeightForType: type];
+      const NSRect        bounds    = [view bounds];
+      const NSRect        frame     = [view frame];
+      NSRect              aRect     = [self tabViewBackgroundRectForBounds: bounds tabViewType: type];
+      NSPoint             iP        = ((type == NSTopTabsBezelBorder) ?
+                                       NSMakePoint(bounds.origin.x, bounds.origin.y) :
+                                       NSMakePoint(rect.origin.x, NSMaxY(aRect)));
+      const NSRect        check     = NSMakeRect(iP.x, iP.y, bounds.size.width, tabHeight);
+      NSDebugMLLog(@"WinNSTabView", @"frame: %@ rect: %@ check: %@ intersection: %@",
+                   NSStringFromRect(frame),
+                   NSStringFromRect(rect), NSStringFromRect(check),
+                   NSStringFromRect(NSIntersectionRect(rect, check)));
+
+      // Avoid drawing tabs if the tab view part is NOT in the requested rectangle...
+      if (NSIntersectsRect(rect, check))
+        {
+          // Using tab class...
+          HTHEME hTheme = [self themeWithClassName: @"tab"];
+
+          if (hTheme == NULL)
+            {
+              // Default to GSTheme...
+              [super drawTabViewRect: rect
+                              inView: view
+                           withItems: items
+                        selectedItem: selected];
+            }
+          else
+            {
+              static const CGFloat  FRAME_ADJUST = 12.0;
+              
+              NSGraphicsContext    *ctxt = GSCurrentContext();
+              const NSTabViewType   type      = [(NSTabView *)view tabViewType];
+              const BOOL            truncate  = [(NSTabView *)view allowsTruncatedLabels];
+              int                   part      = TABP_TABITEMBOTHEDGE;
+              id                    item      = nil;
+              id                    firstItem = [items firstObject];
+              id                    lastItem  = [items lastObject];
+              NSInteger             itemCount = [items count];
+              NSDebugMLLog(@"WinNSTabView", @"rect: %@ aRect: %@",
+                           NSStringFromRect(rect), NSStringFromRect(aRect));
+              
+              // Save the current graphics context state...
+              //DPSgsave(ctxt);
+
+              // Draw the background...
+#if 0
```

Comment:
This sounds strange. Can somebody explain what goes on here and how it should be resolved?